### PR TITLE
Add better default cert/bundle paths for Windows, and de-dupe paths out of ssl_* classes to utils

### PR DESCRIPTION
**Pull request by mitchcapper**
Broken out from #280 and rebased

### CODE REVIEW

File: libwget/utils.c [hunk 1]
```diff
@@ -62,6 +63,37 @@
  * They may be useful to other developers that is why they are exported.
  */
 
+static const char * _wget_ssl_default_path(BOOL bundleNotDir)
```
Comment (rockdaboot) [inline libwget/utils.c:66]: Use `bool` here - needs to include `<stdbool.h>`.

File: libwget/utils.c [hunk 2]
```diff
@@ -62,6 +63,37 @@
...
+		if (!progData)
+			progData = "/ProgramData";
+		const char* dir_separator_str = (char[]){ DIR_SEPARATOR, '\0' };
+		sprintf_s(buffer, MAX_PATH, "%s%s%s%c%s%s", progData, ISSLASH(progData[strlen(progData - 1)]) ? "" : dir_separator_str, "ssl", DIR_SEPARATOR, bundleNotDir ? "ca-bundle.pem" : "certs", bundleNotDir ? "" : dir_separator_str);
+	}
+	return buffer;
+#endif
+}
+const char* wget_ssl_default_cert_dir() {
+	return _wget_ssl_default_path(FALSE);
```
Comment (rockdaboot) [inline libwget/utils.c:89]: Use lowercase `true` here.

File: libwget/utils.c [hunk 3]
```diff
@@ -62,6 +63,37 @@
...
+}
+
+const char* wget_ssl_default_ca_bundle_path()
+{
+	return _wget_ssl_default_path(TRUE);
```
Comment (rockdaboot) [inline libwget/utils.c:94]: Use lowercase `true` here.

File: libwget/utils.c [hunk 4]
```diff
@@ -62,6 +63,37 @@
...
+{
+#ifndef _WIN32
+	return bundleNotDir ? NULL : "/etc/ssl/certs";
+#else
+	static char CERTDIR_PATH[MAX_PATH] = {0};
+	static char CERTBUNDLE_PATH[MAX_PATH] = { 0 };
+	char* buffer = bundleNotDir ? CERTBUNDLE_PATH : CERTDIR_PATH;
+	if (buffer[0]) //we could end up with a partial or incorrect path with multiple threads
+		return buffer;
+	strcpy_s(buffer, MAX_PATH, "/etc/ssl/certs");
```
Comment (rockdaboot) [inline libwget/utils.c:76]: Can we use `wget_strscpy()` here ?

File: libwget/utils.c (hunk 1 above)
Comment (rockdaboot) [inline libwget/utils.c:66]: Please rename to `ssl_default_path`, no need for name-spacing with "wget_", and a leading underscore is reserved in C.

File: libwget/utils.c [hunk 5]
```diff
@@ -62,6 +63,37 @@
  * They may be useful to other developers that is why they are exported.
  */
 
+static const char * _wget_ssl_default_path(BOOL bundleNotDir)
+{
+#ifndef _WIN32
+	return bundleNotDir ? NULL : "/etc/ssl/certs";
+#else
+	static char CERTDIR_PATH[MAX_PATH] = {0};
```
Comment (rockdaboot) [inline libwget/utils.c:71]: A library function should not use `static` variables - except they are only initialized once during initialization phase.
Can we consider `getenv("ProgramData")` to be a constant ? If not, we should better pass this as a function argument. If yes, the function can be called just once in `tls_init()`.

File: src/options.c [hunk 6]
```diff
@@ -1223,6 +1223,9 @@ static int print_plugin_help(WGET_GCC_UNUSED option_t opt,
 }
 
 // default values for config options (if not 0 or NULL)
+/*
```
Comment (rockdaboot) [inline src/options.c:1226]: Please use a C++ style comment here, just for consistency.

File: libwget/utils.c (hunk 5 above)
Comment (mitchcapper) [reply libwget/utils.c:71]: I think assuming programdata is a a constant is just fine, especially as I don't think anyone expects to be able to externally edit the ENV vars of a running program.   I favored initializing on demand as even if called simultaneously there was no race condition but will move the call to tls_init instead.

**Comment by mitchcapper** [comment]
other changes all made but for the initial path setting is there a place better than tls_init to do so? I can do it there but it means adding a call in all ssl engine files to that.

File: libwget/utils.c [hunk 7]
```diff
@@ -62,6 +64,38 @@
...
 
+static const char * ssl_default_path(bool bundleNotDir)
+{
+#ifndef _WIN32
+	return bundleNotDir ? NULL : "/etc/ssl/certs";
+#else
+	static char CERTDIR_PATH[MAX_PATH] = {0};
+	static char CERTBUNDLE_PATH[MAX_PATH] = { 0 };
+	char* buffer = bundleNotDir ? CERTBUNDLE_PATH : CERTDIR_PATH;
+	if (buffer[0]) //we could end up with a partial or incorrect path with multiple threads
```
Comment (rockdaboot) [inline libwget/utils.c:75]: This check can never be true

Comment (rockdaboot) [reply libwget/utils.c:75]: Nevermind, should go to bed :)

**Comment by rockdaboot** [comment]
I pushed your commit to main, but somewhat amended. Please check if it builds on Windows and works as expected.

**Pull request closed without merging**